ndcube/mixins/plotting.py

```diff
@@ -406,21 +409,32 @@ def _animate_cube_1D(self, plot_axis_index=-1, axes_coordinates=None,
                           ylabel="Data [{0}]".format(data_unit), **kwargs)
         return ax
 
-    def _derive_axes_coordinates(self, axes_coordinates, axes_units):
+    def _derive_axes_coordinates(self, axes_coordinates, axes_units, data_shape, edges=False):
         new_axes_coordinates = []
         new_axes_units = []
         default_labels = []
         default_label_text = ""
         for i, axis_coordinate in enumerate(axes_coordinates):
             # If axis coordinate is None, derive axis values from WCS.
             if axis_coordinate is None:
-                # N.B. This assumes axes are independent.  Fix this before merging!!!
-                new_axis_coordinate = self.axis_world_coords(i)
-                axis_label_text = self.world_axis_physical_types[i]
+                # Fix: We would downscale the dependent data into the shape of the axes.
```

Comment:
Can you elaborate more on this comment?  It doesn't make complete sense to me.

Reply:
Yeah, if there `>=2` dependent axes, we would get a `numpy` array of `ndim``>=2`, so that needs to be downgraded to a `1D` array by taking mean along all `non plotting` axes.

Reply:
OK.  Can you capture that in the above comment so it's clear to us in the future the reason for this code block?

Reply:
I was thinking of writing the comment like this - 
```
# If the dimension of new_axis_coordinate is >2D and not equal to dimension of data, then the dimension of new_axis_coordinate needs to be downscaled to a 1D ndarray by taking mean along all non-plotting axes.
```

Reply:
How about
```# If the new_axis_coordinate is not independent, i.e. dimension is >2D and not equal to dimension of data, then the new_axis_coordinate must be reduced to a 1D ndarray by taking the mean along all non-plotting axes.```